### ISSUE TITLE
Is there a way to use PushViewController from Menu

### ISSUE THREAD

**Issue by vkoppaka** [John-Lluch/SWRevealViewController#644]
Hello,

We are using Right Menu and the menu / reveal all seems to work OK. But, in addition to the right menu we still show iOS native back button to go back to the previous view controller the user is on. We are now trying to push some view controllers from the menu and are unable to do so. Is there anyway to use PushViewController (or equivalent) where we keep the back button and still show the hamburger menu?

The issue we are facing is if we call NavigationController.PushViewController, NavigationController on Menu will be nil because Menu does not have a NavigationController. If we are going with RevealViewController().PushViewController or RevealViewController().PresentViewController we are losing the back button.

Any ideas to approach this will be of great help.

Thanks

**Comment by iDevelopper** [comment]
Yes, push your VC with a navigation controller, then push this navigation controller with SW:

```
UINavigationController *frontVC = (UINavigationController *)self.revealViewController.frontViewController;
[frontVC pushViewController: yourVC animated:NO];
[self.revealViewController pushFrontViewController:frontVC animated:YES];
```

**Comment by iDevelopper** [comment]
A sample written in Swift:

[SWNavTest.zip](https://github.com/John-Lluch/SWRevealViewController/files/587291/SWNavTest.zip)

Hope this helps.

**Comment by GaneshKumarSelvam** [comment]
Hey, @iDevelopper how about for pushing view controller when it is with tab bar?

**Comment by iDevelopper** [comment]
By pushing the tab bar controller itself!

**Comment by GaneshKumarSelvam** [comment]
please give me an example? 

UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
   ViewControllerSlide * vcs=[storyboard instantiateViewControllerWithIdentifier:@"ok"];
    UITabBarController * tab=(UITabBarController *)self.revealViewController.frontViewController;
    [tab.navigationController pushViewController:vcs animated:YES];
    [self.revealViewController pushFrontViewController:tab.navigationController animated:YES];


I'm doing this but getting the blank screen!

Reply me ASAP?

![screen shot 2017-08-30 at 4 18 21 pm](https://user-images.githubusercontent.com/22673703/29869070-04947536-8d9f-11e7-8ed1-2c89c50f9d6d.png)
This is my Storyboard rear is slide menu and a front is tab bar, I need push view controller for slide menu!

**Comment by iDevelopper** [comment]
What did you try? Upload a project and I will have a look!

**Comment by GaneshKumarSelvam** [comment]
[SWTabBar.zip](https://github.com/John-Lluch/SWRevealViewController/files/1263348/SWTabBar.zip)


Please have a look @iDevelopper

**Comment by iDevelopper** [comment]
The front view controller is UITabBarController class, not UINavigationController. Then retrieve the controller for the selected tab and its navigation controller and push:

```
    UITabBarController *tbc = (UITabBarController *)self.revealViewController.frontViewController;
    UINavigationController *nc = tbc.selectedViewController;
    [nc pushViewController:myVC animated:NO];
    [self.revealViewController setFrontViewPosition:FrontViewPositionLeft animated:YES];
```

[SWTabBar.zip](https://github.com/John-Lluch/SWRevealViewController/files/1263472/SWTabBar.zip)

**Comment by GaneshKumarSelvam** [comment]
Superb Thanks @iDevelopper  👍

**Comment by IamAbdulQadar** [comment]
Thank you @iDevelopper really you win my heart . I was stuck with this issue and now with your swift code its working perfectly once again thank you so much :)

**Comment by iDevelopper** [comment]
For the future, as SWRevealViewControler seems not to be maintained, see:

https://github.com/iDevelopper/PBRevealViewController

**Comment by chejarla-venkatesh** [comment]
@iDevelopper This is the best solution I have ever found on internet. Thank you so much for your sample code and solution.

**Comment by iDevelopper** [comment]
@chejarla-venkatesh What? SWTabBar sample or 
https://github.com/iDevelopper/PBRevealViewController?

**Comment by chejarla-venkatesh** [comment]
@iDevelopper SWNavTest..🙂

**Comment by iDevelopper** [comment]
@chejarla-venkatesh Why don't you try PBRevealViewController?

**Comment by chejarla-venkatesh** [comment]
@iDevelopper Now My project is almost done and I don't to disturb it. But I will try it with a sample project and would definitely include in my next project if it's better than SWRevealViewController().

**Comment by chejarla-venkatesh** [comment]
@iDevelopper Now My project is almost done and I don't want to disturb it. But I will try it for a sample demo project and would definitely include in my next project for sure.

**Comment by iDevelopper** [comment]
@chejarla-venkatesh  I do not say it's better but SW seems not to be maintained anymore. One more feature of PB vs SW is the possibility that the side view (left or right) can be shown over the main view.

**Comment by chejarla-venkatesh** [comment]
@iDevelopper  I have just seen project demo on github. It really does looks better than
SW. But as I said I can't make changes to my original project right now..
But I will install it for a demo application.
